inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/starrocks/src/main/java/org/apache/inlong/sort/starrocks/table/sink/table/StarRocksDynamicSinkFunctionV2.java

```diff
@@ -291,19 +302,23 @@ public void initializeState(FunctionInitializationContext functionInitialization
             return;
         }
 
+        String transactionStateName = "starrocks-sink-transaction"
+                + (StringUtils.isNullOrWhitespaceOnly(stateKey) ? "" : "-" + stateKey);
```

Comment:
Using the dynamic parameter "stateKey" as the key value, there is a risk of linear growth in the State size of Checkpoint.